### PR TITLE
Implement M405/M406 to enable/disable the filament sensor

### PR DESCRIPTION
**Pull request by Commod0re**
basic implementation of M405 and M406. It compiles but I have not tested it other than that yet

some people on the Prusa Community Forum facebook group were asking about this and it looked straightforward enough for a first contribution :)

### CODE REVIEW

File: Firmware/Marlin_main.cpp [hunk 1]
```diff
@@ -6387,6 +6387,19 @@ if((eSoundMode==e_SOUND_MODE_LOUD)||(eSoundMode==e_SOUND_MODE_ONCE))
...
+    {
+        fsensor_enable();
+    }
+    break;
+
+    case 406: // M406 Disable Filament Sensor
+    {
+        fsensor_disable();
+    }
+#endif
```
Comment (3d-gussner) [inline Firmware/Marlin_main.cpp:7478]: Missing `break;` at the end of M406

Comment (Commod0re) [reply Firmware/Marlin_main.cpp:7478]: nice catch; fixed!

**Comment by Commod0re** [comment]
revisiting this: I've updated my PR to resolve the merge conflict, maybe this can get looked at again soon

File: Firmware/Marlin_main.cpp [hunk 2]
```diff
@@ -7682,6 +7682,20 @@ SERIAL_PROTOCOLPGM("\n\n");
 	}
 	break;
 
+#ifdef FILAMENT_SENSOR
+    case 405: // M405 Enable Filament Sensor
+    {
+        fsensor_enable();
```
Comment (3d-gussner) [inline Firmware/Marlin_main.cpp]: it is now `fsensor.setEnabled(1);`

File: Firmware/Marlin_main.cpp [hunk 3]
```diff
@@ -7682,6 +7682,20 @@ SERIAL_PROTOCOLPGM("\n\n");
...
+    }
+    break;
+
+    case 406: // M406 Disable Filament Sensor
+    {
+        fsensor_disable();
```
Comment (3d-gussner) [inline Firmware/Marlin_main.cpp]: Use `fsensor.setEnabled(0);`

File: Firmware/Marlin_main.cpp [hunk 4]
```diff
@@ -7682,6 +7682,20 @@ SERIAL_PROTOCOLPGM("\n\n");
 	}
 	break;
 
+#ifdef FILAMENT_SENSOR
```
Comment (3d-gussner) [inline Firmware/Marlin_main.cpp:7449]: Add the doxygen documentation for `M405` and `M406`

similar to
https://github.com/prusa3d/Prusa-Firmware/blob/MK3/Firmware/Marlin_main.cpp#L7423-L7434

**Review by 3d-gussner — CHANGES_REQUESTED** [review]

**Review by DRracer — APPROVED** [review]

**Review by leptun — APPROVED** [review]

**Review by DRracer — APPROVED** [review]

**Pull request merged**